### PR TITLE
DROID-3001 App | Tech | Refactoring - provide space

### PR DESCRIPTION
**Pull request by uburoiubu**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/ui/templates/TemplateFragment.kt [hunk 1]
```diff
@@ -27,7 +29,8 @@ import javax.inject.Inject
 class TemplateFragment : BaseFragment<FragmentTemplateBinding>(R.layout.fragment_template),
     ClipboardInterceptor, View.OnDragListener {
 
-    private val ctx: String get() = arg(CTX_KEY)
+    private val ctx: Id get() = arg(CTX_KEY)
+    private val space: Id get() = argOrNull<Id>(SPACE_KEY).orEmpty()
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/templates/TemplateFragment.kt:33]: why is it argOrNull here?

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 2]
```diff
@@ -613,7 +634,12 @@ class HomeScreenViewModel(
         }
         if (subscriptions.isNotEmpty()) unsubscribe(subscriptions)
 
-        closeObject.stream(widgetObject).collect { status ->
+        closeObject.stream(
+            CloseBlock.Params(
+                widgetObject,
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:639]: target = 
space =

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 3]
```diff
@@ -1683,9 +1709,9 @@ class HomeScreenViewModel(
 
     fun onBackClicked() {
         viewModelScope.launch {
-            openWidgetObjectsHistory.forEach { obj ->
+            openWidgetObjectsHistory.forEach { (obj, space) ->
                 closeObject
-                    .async(obj)
+                    .async(CloseBlock.Params(obj, space))
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:1714]: target = 
space =

**Review by konstantiniiv — APPROVED** [review]

File: app/src/main/java/com/anytypeio/anytype/ui/templates/TemplateFragment.kt (hunk 1 above)
Comment (uburoiubu) [reply app/src/main/java/com/anytypeio/anytype/ui/templates/TemplateFragment.kt:33]: fixed

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt (hunk 2 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:639]: fixed

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt (hunk 3 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt]: fixed

**Pull request merged**